packages/fxa-auth-server/test/local/senders/email.js

```diff
@@ -239,802 +813,171 @@ describe('lib/senders/email:', () => {
     assert.notEqual(mailer.mailer, mailer.emailService);
   });
 
-  messageTypes.forEach(type => {
-    const message = {
-      code: 'abc123',
-      deviceId: 'foo',
-      email: 'a@b.com',
-      locations: [],
-      numberRemaining: 2,
-      productId: 'wibble',
-      service: 'sync',
-      tokenCode: 'abc123',
-      uaBrowser: 'Firefox',
-      uaBrowserVersion: '70.0a1',
-      uid: 'uid',
-      unblockCode: 'AS6334PK',
-      type: 'secondary',
-      flowId:
-        '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef',
-      flowBeginTime: Date.now(),
-    };
-
-    let expectedTemplateName;
-    if (subscriptionTypes.has(type)) {
-      expectedTemplateName = type.substr(0, type.lastIndexOf('Email'));
-    } else {
-      expectedTemplateName = type;
-    }
-
-    if (SUBJECT_ASSERTIONS.has(type)) {
-      it(`set the correct subject for ${type}`, () => {
-        mailer.mailer.sendMail = stubSendMail(emailConfig => {
-          assert.equal(emailConfig.subject, SUBJECT_ASSERTIONS.get(type));
+  it('declarative tests', async () => {
+    for (const [type, test] of TESTS) {
```

Comment:
This is the single, declarative test case that exercises all of the above assertion data.